app/knights.py

```diff
@@ -0,0 +1,24 @@
+class Knights:
+
+    def __init__(self, knight, config):
+        self.knight = knight
+        self.config = config
+        self.name = self.config["name"]
+        self.hp = self.config["hp"] + \
+            (self.config["potion"]["effect"].get("hp", 0)
+                if self.config["potion"] is not None else 0)
+        self.power = self.config["power"] + self.config["weapon"]["power"] + \
+            (self.config["potion"]["effect"].get("power", 0)
+                if self.config["potion"] is not None else 0)
+        self.protection = sum(armour["protection"] for armour
+                              in self.config["armour"]) + \
+            (self.config["potion"]["effect"].get("protection", 0)
+                if self.config["potion"] is not None else 0)

```

Comment:
These code duplications can be fixed, using `stats = ("protection", "power", "hp")` and looping on `stats`

Reply:
I have no idea how to do this :(
Could you help me?